### DISCUSSION TITLE
Install 2 formulas that share the same name?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#2055] [Everyday usage]
I want to install both the `curl` formula in the core tap and the one in a 3rd party tap `cloudflare/homebrew-cloudflare/curl` (`/usr/local/Homebrew/Library/Taps/cloudflare/cloudflare/curl.rb`), but since they share the same name, homebrew refused to install the cloudflare curl when the former one is installed.

Any way to have both of them installed?

**Comment by SMillerDev** [comment]
Not without renaming one of them. Names are the identifying information for formulae.

**Reply by tddschn** [reply]
I noticed that homebrew doesn't support package namespacing, 
all formulas with the same name, no matter which tap (namespace) they're from, 
end up in the same `opt/<formula-name>` dir.

What are the reasons that homebrew is designed this way? 

Supporting installing package in the same name but of different taps is sure a lot of work and a lot messier, 
but it can be useful to many people too (I guess). 

For example, I can deal with multiple `curl` in my `$PATH`. 
The curl in the core tap is very up-to-date while the cloudflare one is built with http/3 support, 
it would be really cool if I can have both of them intsalled with homebrew.
